bftengine/include/bftengine/DbCheckpointManager.hpp

```diff
@@ -32,30 +32,43 @@
 #include "Metrics.hpp"
 #include "InternalBFTClient.hpp"
 #include "storage/db_interface.h"
-#if __has_include(<filesystem>)
-#include <filesystem>
-namespace fs = std::filesystem;
-#elif __has_include(<experimental/filesystem>)
-#include <experimental/filesystem>
-namespace _fs = std::experimental::filesystem;
-#else
-#error "Missing filesystem support"
-#endif
+#include <boost/filesystem.hpp>
+namespace _fs = boost::filesystem;
 namespace bftEngine::impl {
 using std::chrono::duration_cast;
 using Status = concordUtils::Status;
 using SystemClock = std::chrono::system_clock;
 using Seconds = std::chrono::seconds;
 using InternalBftClient = bftEngine::impl::InternalBFTClient;
+using DbCheckpointId = uint64_t;
+using BlockId = uint64_t;
 class DbCheckpointManager {
  public:
-  // void createDbCheckpoint(const SeqNum& seqNum);
-  void createDbCheckpointAsync(const SeqNum& seqNum, const std::optional<Timestamp>& timestamp);
+  /***************************************************
+   *@Input parameter1: Request sequnce number
+   *@Input parameter2: Request timestamp
+   *@Description: Creates rocksdb checkpoint asynchronously if database has new state.
+   *              If we already have a checkpoint with same state as lastBlock in the DB, we reject the request.
+   *              Creating another db checkpoint with the same state as the previous one has no use.
+   *              Note: if ReplicaConfig::maxNumberOfDbCheckpoints is set to zero, then also we do not create rocksDb
+   *              and return std::nullopt in that case
+   *@Return: returns the last blockId at which create db checkpoint request is accepted. Else return std::nullopt
```

Comment:
I think it is better to not be explicit about what the returned checkpoint ID is. We might say `an unique checkpoint ID`.